### PR TITLE
feat: nc external signing method

### PR DESCRIPTION
**Pull request by r4mmer**
### Acceptance Criteria

- Allow externally signed nano contract transactions.

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/nano_contracts/utils.ts [hunk 1]
```diff
@@ -28,24 +27,26 @@ import {
...
+export const signAndPushNCTransaction = async (tx: NanoContract, pin: string, storage: IStorage): Promise<Transaction> => {
+  const txSigData = await storage.getTxSignatures(tx, pin);
+  for (const sigData of txSigData.inputSignatures) {
+    const input = tx.inputs[sigData.inputIndex];
+    const inputData = transactionUtils.createInputData(
+      sigData.signature,
+      sigData.pubkey,
+    );
+    input.setData(inputData);
+  }
```
Comment (pedroferreira1) [inline src/nano_contracts/utils.ts:48]: You have a method in the `transactionUtils` (`signTransaction`) that has precisely this same loop. What do you think about creating a util method that receives the signatures and tx to reuse the code?

Comment (r4mmer) [reply src/nano_contracts/utils.ts]: Changed the `signTransaction` to include the nano_contract specific logic so we can use it here

File: src/new/wallet.js [hunk 2]
```diff
@@ -2718,29 +2718,25 @@ class HathorWallet extends EventEmitter {
       throw new PinRequiredError(ERROR_MESSAGE_PIN_REQUIRED);
     }
 
-    // Get private key that will sign the nano contract transaction
-    let privateKey;
-    try {
-      privateKey = await this.getPrivateKeyFromAddress(address, options);
```
[the remark this answers is not in the document]
Comment (r4mmer) [reply src/new/wallet.js:2724]: There is another util that uses this

**Review by pedroferreira1 — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

File: src/utils/transaction.ts [hunk 3]
```diff
@@ -134,13 +135,25 @@ const transaction = {
...
+   */
+  async getSignatureForTx(tx: Transaction, storage: IStorage, pinCode: string): Promise<ITxSignatureData> {
     const xprivstr = await storage.getMainXPrivKey(pinCode);
     const xprivkey = HDPrivateKey.fromString(xprivstr);
     const dataToSignHash = tx.getDataToSignHash();
     const signatures: IInputSignature[] = [];
+    let ncCallerSignature: Buffer|null = null;
 
     for await (const {tx: spentTx, input, index: inputIndex} of storage.getSpentTxs(tx.inputs)) {
+      if (input.data) {
```
Comment (andreabadesso) [inline src/utils/transaction.ts:152]: codecov is complaining this is not covered by tests, should it be? I feel it is a fundamental part of this method and should fail if this behavior is changed in the future

Comment (r4mmer) [reply src/utils/transaction.ts:152]: Created a test for this

**Review by pedroferreira1 — APPROVED** [review]

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**